CMakeLists.txt

```diff
@@ -66,6 +66,7 @@ option( DETRAY_BUILD_INTEGRATIONTESTS "Build the integration tests of Detray"
    OFF )
 option( DETRAY_BUILD_BENCHMARKS "Build the benchmark tests" OFF  )
 option( DETRAY_BUILD_TUTORIALS "Build the tutorial executables of Detray" OFF )
+cmake_dependent_option(DETRAY_NO_HOST_BUILD "Do not build host targets" OFF "DETRAY_BUILD_CUDA OR DETRAY_BUILD_SYCL" OFF)
```

Comment:
Wouldn't it be nicer to have this `DETRAY_BUILD_HOST` in line with the CUDA and SYCL flags? And then to have it on by default?